### PR TITLE
Fix various small things throughout the codebase

### PR DESCRIPTION
**Pull request by burgerdev**
### Context

While browsing the code base, I stumbled over some minor opportunities that don't justify a dedicated PR each. This is a bulk submission of such findings.

### Proposed change(s)

See also individual commits.

- remove obsolete docs, add fresh docs
- prettify versionsapi-list output
- use explicit Helm version in GH action
- use constants instead of literals
- check integrity of rekor-cli
- make `rm` invocation more robust
- Minor correctness and simplification tweaks.

### Checklist

- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: .github/workflows/build-os-image.yml [hunk 1]
```diff
@@ -575,6 +575,7 @@ jobs:
         shell: bash
         run: |
           curl -fsSLO https://github.com/sigstore/rekor/releases/download/v0.12.0/rekor-cli-linux-amd64
+          echo 23c9baefe12b28879275f5aa84dbe9cd734d67ba42bf9bbb28a3eedbf1ed2dd4  rekor-cli-linux-amd64 | sha256sum -c --strict
```
Comment (malt3) [inline .github/workflows/build-os-image.yml:578]: This is obsolete with #2798, where we install cosign and rekor-cli via nix.

**Review by malt3 — APPROVED** [review]
Nice!

**Review by 3u13r — APPROVED** [review]
LGTM

**Review by daniel-weisse — APPROVED** [review]

File: .github/workflows/build-os-image.yml (hunk 1 above)
Comment (burgerdev) [reply .github/workflows/build-os-image.yml:578]: I'm reverting it because I need to fix conflicts anyway.

**Pull request merged**